### DISCUSSION TITLE
show live track of tracking device without internet connection

### DISCUSSION THREAD

**Discussion by maromme** [osmandapp/OsmAnd#11343] [Q&A]
Hi,
I'm developing a small tracking device sending it's position via lora/gsm to a gateway (esp32 based if of interest). The gateway could be connected via Bluetooth or WLAN to a android smartphone. There I wan't to run osmand to show the current location and track of that tracker. 
But I'm not really sure how that last part could work. My best idea is to run a webserver (http/ftp or similar) on the gateway, have a gpx file constantly updated and osmand anyhow  downloading that gpx every for example every 30sec. But now clue how the last part is working out. 
The most important thing is: i can't rely on having a network connection. Therefore the gateway 

I' m looking forward to you ideas and input!
thanks Markus

**Comment by vshcherb** [comment]
There is an api to refresh and load GPX file in OsmAnd, so probably you can create a small program that will connect via  AIDL

**Reply by maromme** [reply]
Hi thanks  for your answer.  sounds like an good approach.  but unfortunately I'm not at all into android app programming and don'tplan to get into it.  hoped for some solution out of the box...
